qcodes/instrument_drivers/tektronix/Keithley_2450.py

```diff
@@ -108,27 +108,101 @@ def __init__(self, parent: 'Keithley2450', name: str, proper_function: str) -> N
             vals=Numbers(0, 1e4)
         )
 
-    def _measure(self) -> str:
+        self.add_parameter(
+            'auto_zero',
+            get_cmd=f":SENSe:{self._proper_function}:AZERo?",
+            set_cmd=f":SENSe:{self._proper_function}:AZERo {{}}",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0"),
+            docstring="This command enables or disables automatic updates to"
+                      "the internal reference measurements (autozero) of the"
+                      "instrument."
+        )
+
+        self.add_parameter(
+            'auto_zero_once',
+            set_cmd=f":SENSe:AZERo:ONCE",
+            docstring="This command causes the instrument to refresh the"
+                      "reference and zero measurements once"
+        )
+
+        self.add_parameter(
+            'count',
+            get_cmd=f":SENSe:COUNt?",
+            set_cmd=f":SENSe:COUNt {{}}",
+            docstring="The number of measurements to make when a measurement "
+                      "is requested."
+        )
+
+    def fetch(
+            self,
+            buffer_name: str = 'defbuffer1',
+            buffer_elements: Optional[list] = None
+    ) -> str:
+        """
+        This method requests the latest reading from a reading buffer.
+
+        Args:
+            buffer_name: buffer name, default is 'defbuffer1'
+            buffer_elements: available options are "DATE", "FORMatted",
+                "FRACtional", "READing", "RELative", "SEConds",
+                "SOURce", "SOURFORMatted", "SOURSTATus",
+                "SOURUNIT", "STATus","TIME", "TSTamp", "UNIT"
```

Comment:
it'd be great if the driver would check the validity of the "fields"/"elements" that the user is entering here because a typo will result in an inconvenient visa io error, i guess? it'd be better if user was getting an exception with a clear message if he/she mistypes the field/element name.

Reply:
Good point. Now the driver is checking if the user input is correct or not, with feedback.